beacon/goclient/attest.go

```diff
@@ -54,46 +69,339 @@ func (gc *GoClient) GetAttestationData(slot phase0.Slot) (
+) {
+	addr := client.Address()
+	attDataReqStart := time.Now()
+
+	logger = logger.With(zap.String("client_addr", addr))
+
+	logger.Debug("fetching attestation data")
+	response, err := client.AttestationData(ctx, &api.AttestationDataOpts{
+		Slot:           slot,
+		CommitteeIndex: 0,
+	})
+
+	recordRequestDuration(ctx, "AttestationData", addr, http.MethodGet, time.Since(attDataReqStart), err)
+
+	if err != nil {
+		logger.Error(clResponseErrMsg, zap.Error(err))
+		errCh <- &attestationDataError{
+			clientAddr: addr,
+			err:        err,
+		}
+		return
+	}
+	if response == nil {
+		logger.Error(clNilResponseErrMsg)
+		errCh <- &attestationDataError{
+			clientAddr: addr,
+			err:        fmt.Errorf("response is nil"),
+		}
+		return
+	}
+	attestationData := response.Data
+	if attestationData == nil {
+		logger.Error(clNilResponseDataErrMsg)
+		errCh <- &attestationDataError{
+			clientAddr: addr,
+			err:        fmt.Errorf("attestation data nil"),
+		}
+		return
+	}
+
+	logger.Debug("scoring attestation data")
+	score := gc.scoreAttestationData(ctx, attestationData, logger)
+
+	respCh <- &attestationDataResponse{
+		clientAddr:      addr,
+		attestationData: attestationData,
+		score:           score,
+	}
+}
+
+// scoreAttestationData generates a score for attestation data.
+// The score is relative to the reward expected from the contents of the attestation.
+func (gc *GoClient) scoreAttestationData(ctx context.Context,
+	attestationData *phase0.AttestationData,
+	logger *zap.Logger,
+) float64 {
+	logger = logger.With(
+		fields.BlockRoot(attestationData.BeaconBlockRoot),
+		zap.Uint64("attestation_data_slot", uint64(attestationData.Slot)))
+	// Initial score is based on height of source and target epochs.
+	score := float64(attestationData.Source.Epoch + attestationData.Target.Epoch)
+
+	// Increase score based on the nearness of the head slot.
+	slot, err := gc.blockRootToSlot(ctx, attestationData.BeaconBlockRoot, logger)
+	if err != nil {
+		logger.
+			With(zap.Error(err)).
+			Error("failed to obtain slot for block root")
+	} else {
+		score += float64(1) / float64(1+attestationData.Slot-slot)
+	}
+
+	logger.With(
+		zap.Uint64("head_slot", uint64(slot)),
+		zap.Uint64("source_epoch", uint64(attestationData.Source.Epoch)),
+		zap.Uint64("target_epoch", uint64(attestationData.Target.Epoch)),
+		zap.Float64("score", score),
+	).Debug("scored attestation data")
+
+	return score
+}
+
+func (gc *GoClient) blockRootToSlot(ctx context.Context, root phase0.Root, logger *zap.Logger) (phase0.Slot, error) {
+	slot, err, _ := gc.blockRootToSlotReqInflight.Do(root, func() (phase0.Slot, error) {
+		cacheResult := gc.blockRootToSlotCache.Get(root)
+		if cacheResult != nil {
+			cachedSlot := cacheResult.Value()
+			logger.
+				With(zap.Int("cache_len", gc.blockRootToSlotCache.Len())).
+				Debug("obtained slot from cache")
+			return cachedSlot, nil
+		}
+
+		logger.Debug("slot was not found in cache. Fetching from Consensus Client")
+
```

Comment:
to keep our logging consistent (lower case & prefer commas rather than periods)
```suggestion
		logger.Debug("slot was not found in cache, fetching from Consensus Client")

```